waku/waku_sync/raw_bindings.nim

```diff
@@ -6,173 +6,234 @@ else:
 from os import DirSep
 
 import
-  std/[strutils, sequtils]
+   std/[strutils],
+   ../waku_core/message,
+   chronicles,
+   std/options,
+   stew/byteutils
 
-import
-   ../waku_core/message
-
-{.link: "../waku_sync/negentropy.so".} #TODO build the dyn lib
-
-const negentropyPath = currentSourcePath.rsplit(DirSep, 1)[0]
+const negentropyPath = currentSourcePath.rsplit(DirSep, 1)[0] & DirSep & ".." & DirSep & ".." & DirSep & "vendor" & DirSep & "negentropy" & DirSep & "cpp" & DirSep
 
-const NEGENTROPY_HEADER = negentropyPath & DirSep & "headers" & DirSep & "negentropy.h"
+{.link: negentropyPath & "libnegentropy.so".} 
 
-### String ###
+const NEGENTROPY_HEADER = negentropyPath & "negentropy_wrapper.h"
```

Comment:
maybe you want to pass the flags to the linker using passL instead of using this way?

Reply:
This doesn't seem to be the issue for compilation error though, also i fixed the tests for now by copying the library to main nwaku folder after it is built.